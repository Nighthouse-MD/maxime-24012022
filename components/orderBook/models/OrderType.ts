enum OrderType {
    ask = "sell",
    bid = "buy"
}

export default OrderType;