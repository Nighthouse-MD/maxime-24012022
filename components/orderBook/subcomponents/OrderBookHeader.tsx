import { memo } from 'react';
import { Col, Row } from 'react-bootstrap';
import styles from '../OrderBook.module.scss'
import OrderType from '../models/OrderType';

const OrderBookHeader = ({ type }: Props) => {
    return <Row className={`${styles.orderBookHeader} ${type === OrderType.bid ? styles.bidsHeader : ''}`}>
        <Col className={styles.price} md={{ order: type === OrderType.ask ? 1 : 3 }}>
            PRICE
        </Col>
        <Col className={styles.quantity} md={{ order: 2 }}>
            SIZE
        </Col>
        <Col className={styles.quantity} md={{ order: type === OrderType.ask ? 3 : 1 }}>
            TOTAL
        </Col>
    </Row>;
}

interface Props {
    type: OrderType;
}

export default memo(OrderBookHeader);