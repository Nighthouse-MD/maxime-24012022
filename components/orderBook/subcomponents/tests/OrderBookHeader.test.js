import OrderBookHeader from '../OrderBookHeader';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import OrderType from '../../models/OrderType';

describe('OrderBookHeader', () => {
    it('shows the correct order for OrderType.ask', () => {
        render(<OrderBookHeader type={OrderType.ask} />);

        const priceElement = screen.queryByText("PRICE");
        expect(priceElement).toBeInTheDocument();
        expect(priceElement.className).toBe('price order-md-1 col');

        const sizeElement = screen.queryByText("SIZE");
        expect(sizeElement).toBeInTheDocument();
        expect(sizeElement.className).toBe('quantity order-md-2 col');

        const totalElement = screen.queryByText("TOTAL");
        expect(totalElement).toBeInTheDocument();
        expect(totalElement.className).toBe('quantity order-md-3 col');
    });

    it('shows the correct order for OrderType.bid', () => {
        render(<OrderBookHeader type={OrderType.bid} />);

        const priceElement = screen.queryByText("PRICE");
        expect(priceElement).toBeInTheDocument();
        expect(priceElement.className).toBe('price order-md-3 col');

        const sizeElement = screen.queryByText("SIZE");
        expect(sizeElement).toBeInTheDocument();
        expect(sizeElement.className).toBe('quantity order-md-2 col');

        const totalElement = screen.queryByText("TOTAL");
        expect(totalElement).toBeInTheDocument();
        expect(totalElement.className).toBe('quantity order-md-1 col');
    });

    it('renders correctly for OrderType.ask', () => {
        const tree = renderer
            .create(<OrderBookHeader type={OrderType.ask} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly for OrderType.bid', () => {
        const tree = renderer
            .create(<OrderBookHeader type={OrderType.bid} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});