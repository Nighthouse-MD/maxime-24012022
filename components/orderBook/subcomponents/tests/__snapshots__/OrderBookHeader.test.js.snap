// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBookHeader renders correctly for OrderType.ask 1`] = `
<div
  className="orderBookHeader  row"
>
  <div
    className="price order-md-1 col"
  >
    PRICE
  </div>
  <div
    className="quantity order-md-2 col"
  >
    SIZE
  </div>
  <div
    className="quantity order-md-3 col"
  >
    TOTAL
  </div>
</div>
`;

exports[`OrderBookHeader renders correctly for OrderType.bid 1`] = `
<div
  className="orderBookHeader bidsHeader row"
>
  <div
    className="price order-md-3 col"
  >
    PRICE
  </div>
  <div
    className="quantity order-md-2 col"
  >
    SIZE
  </div>
  <div
    className="quantity order-md-1 col"
  >
    TOTAL
  </div>
</div>
`;
